### PR TITLE
chore: add uneccessary_* lints

### PR DESCRIPTION
**Pull request by eseidel**
Adding:
  - unnecessary_null_checks
  - unnecessary_parenthesis
  - unnecessary_raw_strings Each of these found real cases where fixing with the lint improved readability (one could argue about the raw strings).

### CODE REVIEW

**Comment by eseidel** [comment]
Huh.  Will figure out what I broke and upload again, sorry.

File: lib/src/query.dart [hunk 1]
```diff
@@ -351,7 +351,7 @@ class PostgreSQLFormatIdentifier {
 
       final dataTypeString = variableComponents.last;
       try {
-        type = typeStringToCodeMap[dataTypeString]!;
+        type = typeStringToCodeMap[dataTypeString];
```
Comment (osaxma) [inline lib/src/query.dart:354]: The test that is failing is due to this line of code.

Apparently, the code was throwing an error whenever `type` is `null` by relying on the removed null check -- that is, when the null check fails, it throws a format exception. 

I think a proper way to handle this is to replace the try/catch statement with if/throw such as:

```dart
      type = typeStringToCodeMap[dataTypeString];
      if (type == null) {
        throw FormatException(
            "Invalid type code in substitution variable '$t'");
      }
```

Comment (isoos) [reply lib/src/query.dart:354]: This looks good to me. @eseidel could you please rebase the PR and also fix this code?

**Comment by eseidel** [comment]
Sorry for the delay, I believe this is ready now and should pass CI. 🤞

File: lib/src/query.dart [hunk 2]
```diff
@@ -351,7 +351,11 @@ class PostgreSQLFormatIdentifier {
...
+        if (type == null) {
+          throw FormatException(
+              "Invalid type code in substitution variable '$t'");
+        }
       } catch (e) {
         throw FormatException(
             "Invalid type code in substitution variable '$t'");
```
Comment (osaxma) [inline lib/src/query.dart:361]: The try/catch encapsulating if/throw is redundant and could be removed.

**Comment by eseidel** [comment]
Thanks.  Apologies again.  Got distracted with non-hacking matters. :)

**Pull request merged**